exercises/diffie-hellman/canonical-data.json

```diff
@@ -0,0 +1,78 @@
+{ "exercise": "diffie-hellman"
+, "version" : "1.0.0"
+, "comments":
+    [ " In the following descriptions, p and g represent two          "
+    , " prime numbers.                                                "
+    , "                                                               "
+    , " Unless specified otherwise, tests are setup with              "
+    , " new DiffieHellman object using p = 23; g = 5.                 "
+    , "                                                               "
+    , " Alice's private key = 6                                       "
+    , " Alice's public key = 8                                        "
+    , "                                                               "
+    , " Bob's private key = 15                                        "
+    , " Bob's public key = 19                                         "
+    ]
+, "cases":
+    [ { "description": "Private keys are generated in Range (1..p)"
+      , "property"   : "generatePrivateKey"
+      , "p": 23
+      }
+    , { "description": "generatePrivateKey's output is random."
+      , "property"   : "generatePrivateKey"
```

Comment:
this is difficult to test, but since `property` tells us how we should interpret the other keys in the same object, and we have two cases with same `property` but testing different things, perhaps they should be combined into one, OR this one should have some `property` like `generatePrivateKeyIsRandom`.

Note that there is no precedent for this, since `robot-name` and `simple-cipher` are to my knowledge the only other exercises that need randomness. the former has no JSON file but the latter is https://github.com/exercism/problem-specifications/pull/966, so perhaps the same approach may be used for both.